Dockerfile

```diff
@@ -17,9 +17,11 @@ COPY main.go main.go
 COPY --link api/ api/
 COPY --link controllers/ controllers/
 
+ARG TARGETARCH
+ARG TARGETOS
 # Build
 RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build \
-    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o manager main.go
+    CGO_ENABLED=0 GOOS=$TARGETOS  GOARCH=$TARGETARCH GO111MODULE=on go build -o manager main.go
```

Comment:
are there any targetOS we dont support? did u run this and double check it on your windows pc?

Reply:
I do believe that we don't support windows, but I did this in wsl2.  This is the same thing as the docker-bake.json that we currently have.  It just does this in the Dockerfile.  I needed this to push up the image but I did that already.  I can remove this and only update the image to 1.21.3.  _(Because we do this in a pipeline, it may be better to not do this if windows isn't supported)_

So when building it, I did this: 
`docker buildx build --push --platform linux/arm64/v8,linux/amd64 -t ghcr.io/getporter/porter-operator:v1.0.0 .` so that could get us the image we needed at v1.0.0 which was having issues because the "platform" was missing on the docker manifest.

Reply:
sgtm, we should note what OSes we support here maybe

Reply:
i dont have a strong opinion on this - im just trying to protect against future confusion

Reply:
Agree, I can remove this and allow the image to source it's contents from what is supported.  I initially only needed this to push the operator image to ghcr.  I have done this and can remove this.